modules/docs/sc_storage.md

```diff
@@ -0,0 +1,526 @@
+    - [show: parameters](#show-parameters)
+    - [show: returns](#show-returns)
+    - [show: example](#show-example)
+  - [clear method](#clear-method)
+    - [clear: returns](#clear-returns)
+    - [clear: example](#clear-example)
+  - [get\_properties\_for\_object\_type method](#get_properties_for_object_type-method)
+    - [get\_properties\_for\_object\_type: parameters](#get_properties_for_object_type-parameters)
+    - [get\_properties\_for\_object\_type: returns](#get_properties_for_object_type-returns)
+    - [get\_properties\_for\_object\_type: example](#get_properties_for_object_type-example)
+
+## Introduction
+
+The sc_storage module provides methods to help communicate with storage backends. It has been made in OOP (object oriented programming)
+
+## What can you store
+
+The storage mechanism will only allow to store valid objects. Valid objects are referred as **"storage_objects"** in the code and are defined within the code. They must match one of the following Lua pattern:
+
+- "host_%d+",
+- "service_%d+_%d+",
+- "ba_%d+",
+- "metric_.*"
+
+The above pattern is then called an object_id (host_2712 is a storage object id)
+This rule is here to enforce a readable storage and easily understand which data belongs to what.
+
+## Memory table A.K.A magic table
+
+This feature is a kind of abstraction layer for the storage mechanism.
+When used, it will usually do something in memory but also do something with the persistent storage. (I rarely write sentences that meaningless but I will explain)
+
+### Use case
+
+Usually when you want to store a value in memory, you just put it in a table.
+If you want some persistent storage you also need to use the appropriate function from the sc_storage module.
+
+The memory table is designed to avoid this double work.
+When you set a value inside the memory table, it is automatically going to also set it in the persistent storage.
+
+### How does it work
+
+#### First time adding data in the memory table
+
+First, you have access to a memory table after having initiated the sc_storage module
+
+```lua
+local test_storage = sc_storage.new(test_common, test_logger, params)
+
+-- test_storage.memory is the memory table
+```
+
+Then you can store data inside it, let's do it for the first time
+
+```lua
+local object_id = "host_2712"
+local object_properties = {
+  town = "bordeaux",
+  zip_code = 33000
+}
+
+test_storage.memory[object_id] = object_properties
+--> test_storage.memory has now the following structure: 
+--[[
+  test_storage.memory = {
+    host_2712 = {
+      _internal_object_id = "host_2712",
+      town = "bordeaux",
+      zip_code = 33000
+    }
+  }
+]]
+```
+
+As you can see, upon creation, a `_internal_object_id` index has been added in the memory table. This is because it is required by the mechanism in order to know how to store data in the persistent storage. Because while the memory table has been populated with some values, so does the persistent storage.
```

Comment:
```suggestion
As you can see, upon creation, an `_internal_object_id` index has been added to the memory table. This is because it is required by the mechanism in order to know how to store data in the persistent storage. Because while the memory table has been populated with some values, so has the persistent storage.
```